client/src/main/java/io/split/engine/matchers/BooleanMatcher.java

```diff
@@ -0,0 +1,46 @@
+package io.split.engine.matchers;
+
+import io.split.client.SplitClientImpl;
+
+import java.util.Map;
+
+import static io.split.engine.matchers.Transformers.asBoolean;
+
+public class BooleanMatcher implements Matcher {
+    private boolean _booleanValue;
+
+    public BooleanMatcher(boolean booleanValue) {
+        _booleanValue = booleanValue;
+    }
+
+    @Override
+    public boolean match(Object matchValue, String bucketingKey, Map<String, Object> attributes, SplitClientImpl splitClient) {
+        if (matchValue == null) {
+            return false;
+        }
+
+        Boolean valueAsBoolean = asBoolean(matchValue);
+
+        return valueAsBoolean != null && valueAsBoolean == _booleanValue;
+    }
+
+    @Override
+    public String toString() {
+        return "is " + Boolean.toString(_booleanValue);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        BooleanMatcher that = (BooleanMatcher) o;
+
+        return _booleanValue == that._booleanValue;
```

Comment:
Unboxing for equality checks is unnecessary, Java does it automatically: https://stackoverflow.com/questions/30454635/when-using-for-a-primitive-and-a-boxed-value-is-autoboxing-done-or-is-unbox